### ISSUE TITLE
gulp tasks failing with Error: jquery is not installed.

### ISSUE THREAD

**Issue by shubhojitroy3** [johnpapa/generator-hottowel#167]
After generating a new app I tried running gulp build.

Initially had the Error: Error: Cannot find where you keep your Bower packages.
Thus followed solutions mentioned in 
https://github.com/johnpapa/generator-hottowel/issues/137
and
https://github.com/johnpapa/generator-hottowel/issues/156

After that, when running gulp build, got the following error

```
Error: Error: jquery is not installed. Try running `bower install`.
    at E:\Workspace\Unit Tests\papafiedMocha\node_modules\wiredep\wiredep.js:30:56
    at E:\Workspace\Unit Tests\papafiedMocha\node_modules\wiredep\lib\detect-dependencies.js:145:29
    at forOwn (E:\Workspace\Unit Tests\papafiedMocha\node_modules\wiredep\node_modules\lodash\dist\lodash.js:2106:15)
    at Function.forEach (E:\Workspace\Unit Tests\papafiedMocha\node_modules\wiredep\node_modules\lodash\dist\lodash.js:3303:9)
    at detectDependencies (E:\Workspace\Unit Tests\papafiedMocha\node_modules\wiredep\lib\detect-dependencies.js:34:7)

    at wiredep (E:\Workspace\Unit Tests\papafiedMocha\node_modules\wiredep\wiredep.js:70:39)
    at module.exports (E:\Workspace\Unit Tests\papafiedMocha\gulp.config.js:10:22)
    at Object.<anonymous> (E:\Workspace\Unit Tests\papafiedMocha\gulpfile.js:3:38)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
```

I did reinstall jQuery using bower
`bower install jQuery`

But, I'm still getting the same error. 
Anyone has any idea what might be the issue here?

**Comment by johnpapa** [comment]
unable to repro. have you tried the latest?

**Comment by shubhojitroy3** [comment]
My apologies for the delay in responding.
Had temporarily moved to other assignments. 
Will try a fresh install again over this weekend and update here.

**Comment by dashawk** [comment]
Try installing jQuery with bower like this
`bower install jquery --save`


Hope this helps.

**Comment by jkruer01** [comment]
I'm getting the same error.  I tried bower install jquery --save but that did not resolve the issue.

**Comment by dashawk** [comment]
Did you try deleting the bower_components folder? then run `bower install`?

**Comment by jkruer01** [comment]
Yes I tried that and it still did not work.  While I was doing the bower install, it asked me the following question and I chose #1:

```
Unable to find a suitable version for jquery, please choose one by typing one of the numbers below:
    1) jquery#>=1.6.3 <3 which resolved to 2.2.4 and is required by toastr#2.1.3
    2) jquery#^3.1.1 which resolved to 3.1.1 and is required by value-investing-mentor
    3) jquery#>= 1.9.0 which resolved to 3.1.1 and is required by bootstrap#3.2.0

Prefix the choice with ! to persist it to bower.json

? Answer 1
```

Should I have chose a different version?

**Comment by dashawk** [comment]
you can choose whatever version you like but I prefer to have the latest version. While choosing, persist your choice by adding an exclamation point after your answer then press enter to save it in your bower.json file.

**Comment by jkruer01** [comment]
I deleted jquery and ran bower install.  I selected the latest version and used the ! after it.  Then I ran gulp build and I still got the same jquery error:

```
c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web>bower install
bower jquery#>=1.6.3 <3         cached https://github.com/jquery/jquery-dist.git#2.2.4
bower jquery#>=1.6.3 <3       validate 2.2.4 against https://github.com/jquery/jquery-dist.git#>=1.6.3 <3
bower jquery#>= 1.9.0           cached https://github.com/jquery/jquery-dist.git#3.1.1
bower jquery#>= 1.9.0         validate 3.1.1 against https://github.com/jquery/jquery-dist.git#>= 1.9.0
bower jquery#^3.1.1             cached https://github.com/jquery/jquery-dist.git#3.1.1
bower jquery#^3.1.1           validate 3.1.1 against https://github.com/jquery/jquery-dist.git#^3.1.1

Unable to find a suitable version for jquery, please choose one by typing one of the numbers below:
    1) jquery#>=1.6.3 <3 which resolved to 2.2.4 and is required by toastr#2.1.3
    2) jquery#>= 1.9.0 which resolved to 3.1.1 and is required by bootstrap#3.2.0
    3) jquery#^3.1.1 which resolved to 3.1.1 and is required by value-investing-mentor

Prefix the choice with ! to persist it to bower.json

? Answer 3!
bower jquery                resolution Saved jquery#^3.1.1 as resolution
bower jquery#^3.1.1            install jquery#3.1.1

jquery#3.1.1 wwwroot\lib\jquery

c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web>gulp build
[07:36:30] Using gulpfile c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\gulpfile.js
[07:36:30] Starting 'webConfig'...
[07:36:30] Starting 'wiredep'...
[07:36:30] Wiring the bower dependencies into the html
[07:36:30] Starting 'clean-styles'...
[07:36:30] Cleaning: ./.tmp/**/*.css,./wwwroot/styles/**/*.css
[07:36:30] Starting 'clean-code'...
[07:36:30] Cleaning: ./.tmp/**/*.js,./wwwroot/js/**/*.js,./wwwroot/**/*.html
[07:36:30] Starting 'vet'...
[07:36:30] Analyzing source with JSHint and JSCS
[07:36:31] Starting 'clean-images'...
[07:36:31] Cleaning: ./wwwroot/images/**/*.*
[07:36:31] Starting 'clean-fonts'...
[07:36:31] Cleaning: ./wwwroot/fonts/**/*.*
[07:36:31] Finished 'clean-images' after 13 ms
[07:36:31] Starting 'images'...
[07:36:31] Compressing and copying images
[07:36:31] Finished 'clean-fonts' after 684 ms
[07:36:31] Starting 'bootstrap-fonts'...
[07:36:31] Copying bootstrap fonts
[07:36:31] Finished 'clean-styles' after 1.65 s
[07:36:31] Starting 'styles'...
[07:36:31] Compiling Less --> CSS
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Error: jquery is not installed. Try running `bower install`.
    at c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\wiredep.js:30:56
    at c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\lib\detect-dependencies.js:145:29
    at forOwn (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\lodash\dist\lodash.js:2106:15)
    at Function.forEach (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\lodash\dist\lodash.js:3303:9)
    at detectDependencies (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\lib\detect-dependencies.js:34:7)
    at wiredep (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\wiredep.js:70:39)
    at DestroyableTransform._transform (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\wiredep.js:135:34)
    at DestroyableTransform.Transform._read (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\node_modules\through2\n
ode_modules\readable-stream\lib\_stream_transform.js:184:10)
    at DestroyableTransform.Transform._write (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\node_modules\through2\
node_modules\readable-stream\lib\_stream_transform.js:172:12)
    at doWrite (c:\Src\valueinvestingmentor-netcore\ValueInvestingMentor.Web\node_modules\wiredep\node_modules\through2\node_modules\readable-stream\l
ib\_stream_writable.js:237:10)
```

**Comment by dashawk** [comment]
Hi, sorry for the late reply.

If this issue is still open, I have at least few more scenarios on this problem.

- If you are using a windows machine, probably your node_modules package is located somewhere else where you do not have access with.
- If you are using Mac or Linux OS, you might not have the permission to install packages.

--------------------------------------------------------------------------
 -> Windows = run your cmd or bash terminal in administrator mode then run `bower install jquery --save` again.
-> Linux/Mac = run with sudo `sudo bower install jquery --save`.

Also, after you successfully installed jquery,  you might want to add it in the constants.js file
```js
angular
    .module('app.core')
    .constant('toastr', toastr)
    .constant('$', window.jQuery)
    .constant('moment', moment);
```

Hope this works.
